src/apm/health.go

```diff
@@ -0,0 +1,185 @@
+	container := &pod.Spec.Containers[firstContainer]
+
+	err := validateContainerEnv(container.Env, envHealthFleetControlFilepath)
+	if err != nil {
+		return *originalPod, err
+	}
+
+	var sidecarContainerEnv []corev1.EnvVar
+
+	// copy the env var for the health filepath
+	if idx := getIndexOfEnv(container.Env, envHealthFleetControlFilepath); idx > -1 {
+		sidecarContainerEnv = append(sidecarContainerEnv, container.Env[idx])
+	}
+
+	i.injectEnvVarsIntoTargetedEnvVars(inst.Spec.HealthAgent.Env, &container.Env)
+	i.injectEnvVarsIntoSidecarEnvVars(inst.Spec.HealthAgent.Env, &sidecarContainerEnv)
+
+	var healthMountPath string
+	{
+		v, _ := getValueFromEnv(container.Env, envHealthFleetControlFilepath)
+		if healthMountPath, err = i.validateHealthFilepath(v); err != nil {
+			return *originalPod, fmt.Errorf("invalid env value %q for %q > %w", v, envHealthFleetControlFilepath, err)
+		}
+	}
+
+	// set defaults
+	for _, entry := range healthDefaultEnv {
+		if idx := getIndexOfEnv(sidecarContainerEnv, entry.Name); idx == -1 {
+			sidecarContainerEnv = append(sidecarContainerEnv, corev1.EnvVar{
+				Name:  entry.Name,
+				Value: entry.Value,
+			})
+		}
+	}
+
+	sidecarListenPort := defaultHealthListenPort
+	if v, ok := getValueFromEnv(sidecarContainerEnv, envHealthListenPort); ok {
+		sidecarListenPort, err = i.validateHealthListenPort(v)
+		if err != nil {
+			return *originalPod, fmt.Errorf("invalid env value %q for %q > %w", v, envHealthListenPort, err)
+		}
+	}
+
+	if isContainerVolumeMissing(&pod.Spec.Containers[firstContainer], healthVolumeName) {
+		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+			Name:      healthVolumeName,
+			MountPath: healthMountPath,
+		})
+	}
+
+	// We just inject Volumes and init containers for the first processed container.
+	if isInitContainerMissing(pod, healthSidecarContainerName) {
+		if isPodVolumeMissing(pod, healthVolumeName) {
+			pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
+				Name: healthVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				}})
+		}
+
+		restartAlways := corev1.ContainerRestartPolicyAlways
+		sidecarContainer := corev1.Container{
+			Name:          healthSidecarContainerName,
+			Image:         inst.Spec.Agent.Image,
+			RestartPolicy: &restartAlways,
+			VolumeMounts: []corev1.VolumeMount{{
+				Name:      healthVolumeName,
+				MountPath: healthMountPath,
+			}},
+			Ports: []corev1.ContainerPort{
+				{ContainerPort: int32(sidecarListenPort)},
+			},
+			Env: sidecarContainerEnv,
+		}
+
+		pod.Spec.InitContainers = append(pod.Spec.InitContainers, sidecarContainer)
+	}
+
+	if pod.Annotations == nil {
+		pod.Annotations = map[string]string{}
+	}
+	pod.Annotations[HealthInstrumentedAnnotation] = "true"
+
+	return pod, nil
+}
+
+func (i *baseInjector) injectEnvVarsIntoTargetedEnvVars(instEnvVars []corev1.EnvVar, containerEnvVars *[]corev1.EnvVar) {
```

Comment:
I rarely have seen passing pointer to the slice. Why not make it simple and just do 

`func (i *baseInjector) injectEnvVarsIntoTargetedEnvVars(instEnvVars []corev1.EnvVar, containerEnvVars []corev1.EnvVar) []corev1.EnvVar) {
for _, env := range instEnvVars {
		if env.Name == envHealthFleetControlFilepath {
			if idx := getIndexOfEnv(containerEnvVars, env.Name); idx == -1 {
				containerEnvVars = append(containerEnvVars, env)
			}
			break
		}
	}
	return containerEnvVars
`
Also, why are we breaking here?